lib/fastlane/plugin/run_tests_firebase_testlab/actions/run_tests_firebase_testlab_action.rb

```diff
@@ -1,47 +1,196 @@
+                                       env_name: "BUCKET_URL",
+                                       description: "The bucket url where the test results were stored. Default: Parsed automatically from tests output",
+                                       is_string: true,
+                                       optional: true,
+                                       default_value: nil),
+          FastlaneCore::ConfigItem.new(key: :delete_firebase_files,
+                                       env_name: "DELETE_FIREBASE_FILES",
+                                       description: "A flag to controll if the firebase files should be deleted from the bucket or not. Default: false",
+                                       is_string: false,
+                                       optional: true,
+                                       default_value: false),
+          FastlaneCore::ConfigItem.new(key: :extra_options,
+                                       env_name: "EXTRA_OPTIONS",
+                                       description: "Extra options that you need to pass to the gcloud command. Default: empty string",
+                                       is_string: true,
+                                       optional: true,
+                                       default_value: ""),
+          FastlaneCore::ConfigItem.new(key: :gcloud_service_key_file,
+                                       env_name: "GCLOUD_SERVICE_KEY_FILE",
+                                       description: "File path containing the gcloud auth key. Default: Created from GCLOUD_SERVICE_KEY environment variable",
+                                       is_string: true,
+                                       optional: true,
+                                       default_value: nil)
         ]
       end
 
       def self.is_supported?(platform)
-        # Adjust this if your plugin only works for a particular platform (iOS vs. Android, for example)
-        # See: https://docs.fastlane.tools/advanced/#control-configuration-by-lane-and-by-platform
-        #
-        # [:ios, :mac, :android].include?(platform)
-        true
+        platform == :android
       end
+
+      def self.example_code
+        [
+          'run_tests_firebase_testlab(
+              project_id: "your-firebase-project-id",
+              model: "Nexus6P",
+              version: "27",
+              delete_firebase_files: true
+          )'
+        ]
+      end
+
+      def self.category
+        :testing
+      end
+
+      def self.scrape_bucket_url
+        File.open(@test_console_output_file).each do |line|
+          url = line.scan(/\[(.*)\]/).last&.first
+          next unless !url.nil? and (!url.empty? and url.include?("test-lab-"))
+          splitted_url = url.split("/")
+          length = splitted_url.length
+          return "gs://#{splitted_url[length - 2]}/#{splitted_url[length - 1]}"
+        end
+      end
+
+      private_class_method :scrape_bucket_url
```

Comment:
What is the difference between "private_class_method" and "private" like we did on the Short Stories API? Shouldn't this be defined in the beginning of the class? It's easier to be overlooked in the bottom.

Reply:
private doesn't work with method classes. we need to define the private class method like that, you have an example here: http://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Lint/IneffectiveAccessModifier

I would prefer to define it above the method or in the line below of the end (without an empty line dividing it) but this looks the normal approach. So, I will keep it like this for now.